### ISSUE TITLE
Workspace creation from WIT in che 7 without stackIDs

### ISSUE THREAD

**Issue by sbose78** [openshiftio/openshift.io#4536]
Given that we are moving way from Stack/StackIDs in Che 7, what would be the ideal way to trigger workspace creation from api.openshift.io ?

**Comment by sbose78** [comment]
Need your inputs @l0rd .

**Comment by l0rd** [comment]
Che 7 based workspaces are still working with stacks IDs. What has changed is that a Che 7 stack corresponds to a list of Che plugins (e.g. `java + git + terminal + ...`) rather than a container image (e.g. `centos-jdk8`).

We may in the future discontinue the usage of a stack ID to start a workspace in favor of a list of plugins IDs but that's something we haven't planned for the near future.

**Comment by gorkem** [comment]
We can not sustain stack_ids forever. The import flow will _hopefully_ introduce more unpredictable application architectures, which we will not have a corresponding stack id.

**Comment by ibuziuk** [comment]
Currently I do not understand what would be the openshift.io flow for che 7 **without** support of `stack_id`. For example, it is not clear how the plugins are going to be defined on UI side and passed to wit / che-starter when workspace is created (is it plan to guess set of plugins somehow based on the project type?). 

> The import flow will hopefully introduce more unpredictable application architectures, which we will not have a corresponding stack id.

Are there are any mockups for this new flow ?

**Comment by sbose78** [comment]
I really like the idea of passing a list of plugins. Today, any change in plugin list means a new stack would be needed which doesn't scale.

> Are there are any mockups for this new flow ?


My immediate use case ( pain point ) is that 
- user imports node.js project
- starts che workspace
- `run` command doesn't work ( https://github.com/redhat-developer/rh-che/pull/1078 )

The `run` command might not work because the imported project probably has the `app.js` somewhere else other than the standard location. I might want to on-the-fly define the commands based on a detection mechanism coupled with user input.

In short, it would be great if we could call a `Create workspace` API with a payload containing a list of plugins, commands and project specific decisions.

**Comment by sbose78** [comment]
> is it plan to guess set of plugins somehow based on the project type?

1. Have a reasonable list of default list of plugins for maven|node|golang projects
2. override certain defaults by asking for user inputs in the import flow.

**Comment by ibuziuk** [comment]
@sbose78 what do you think about closing this issue as outdated / won't fix ?

**Comment by ibuziuk** [comment]
@sbose78 @jarifibrahim Closing as won't fix. Feel free to reopen with details
